subprojects/core/src/main/java/org/gradle/internal/problems/DefaultProblemLocationAnalyzer.java

```diff
@@ -116,6 +116,15 @@ public Location locationForUsage(Failure failure, boolean fromException) {
         }
     }
 
+    private static int getStartPosWithLocation(Failure failure) {
+        int startPos = -1;
+        List<StackTraceElement> stackTrace = failure.getStackTrace();
+        do {
+            startPos = failure.indexOfStackFrame(startPos + 1, StackFramePredicate.USER_CODE);
```

Comment:
🤔  The call to `indexOfStackFrame` can return `-1` and it will break the while condition check